src/commitlint/index.spec.ts

```diff
@@ -44,24 +69,84 @@ describe('@co-it/schematics:commitlint', () => {
     });
   });
 
-  describe('When package.json has no husky entry', () => {
+  describe('When .huskyrc does not exist', () => {
     beforeEach(() => {
       const packageBeforeInstall = { devDependencies: {} };
       treeBefore.create('package.json', JSON.stringify(packageBeforeInstall));
     });
 
-    it('should add an husky entry with commitlint', () => {
+    it('should configure a commit-msg hook in .huskyrc', () => {
       const treeAfter = runner.runSchematic('commitlint', {}, treeBefore);
 
-      const packageAfterInstall = JSON.parse(
-        treeAfter.readContent('package.json')
+      const huskyRc = JSON.parse(treeAfter.readContent('.huskyrc'));
+
+      expect(huskyRc).toEqual({
+        hooks: {
+          'commit-msg': 'commitlint -E HUSKY_GIT_PARAMS'
+        }
+      });
+    });
+  });
+
+  describe('When .huskyrc already exists', () => {
+    beforeEach(() => {
+      treeBefore.create(
+        'package.json',
+        JSON.stringify({ devDependencies: {} })
+      );
+      treeBefore.create(
+        '.huskyrc',
+        JSON.stringify({ hooks: { 'pre-commit': 'script' } })
       );
+    });
 
-      expect(packageAfterInstall.husky).toEqual({
+    it('should merge commit-msg hook into .huskyrc', () => {
+      const treeAfter = runner.runSchematic('commitlint', {}, treeBefore);
+
+      const huskyRc = JSON.parse(treeAfter.readContent('.huskyrc'));
+
+      expect(huskyRc).toEqual({
         hooks: {
+          'pre-commit': 'script',
           'commit-msg': 'commitlint -E HUSKY_GIT_PARAMS'
         }
       });
     });
   });
+
+  describe('When a competing husky configuration is found', () => {
+    let mockLogger: { warn: () => void };
+    let warn: jest.SpyInstance;
+
+    beforeEach(() => {
+      mockLogger = { warn: () => {} };
+      runner['_logger'] = { createChild: () => mockLogger as any } as any;
+      warn = jest.spyOn(mockLogger, 'warn');
+
+      const packageBeforeInstall = {
+        scripts: {},
+        devDependencies: {},
+        husky: {}
+      };
+      treeBefore.create('package.json', JSON.stringify(packageBeforeInstall));
+    });
+
+    it('should warn if detected in package.json', () => {
+      runner.runSchematic('commitlint', {}, treeBefore);
+      expect(warn).toHaveBeenCalledWith(
+        'Found competing husky configuration in package.json.'
+      );
+    });
+
+    it.each([['.huskyrc.json'], ['.huskyrc.js']])(
```

Comment:
We double our tests for `prettier` and `commitlint`.
Maybe we find a way to refactor this, too.

In .NET `Mspec` (Testing Framework) there are `Behaviours`.
This mechanism allows for refactoring similar tests.